src/preact-render-spy.js

```diff
@@ -264,58 +264,62 @@ class FindWrapper {
   }
 }
 
+const setHiddenProp = (object, prop, value) => {
+  Object.defineProperty(object, prop, {
+    enumerable: false,
+    configurable: true,
+    value,
+  });
+  return value;
+};
+
+class ContextRootWrapper extends Component {
+  constructor({ vdom, context }) {
+    super({ vdom, context });
+    this.state = {vdom};
+
+    // setup the re-renderer on the RenderContext
+    setHiddenProp(context, 'contextRender', vdom => {
+      this.setState({ vdom });
+      rerender();
+    });
+  }
+
+  render({}, {vdom}) {
+    return vdom;
+  }
+}
+
 class RenderContext extends FindWrapper {
   constructor({depth}) {
     super(null, []);
 
-    Object.defineProperties(this, {
-      context: {
-        value: this,
-        configurable: true,
-        enumerable: false,
-      },
-      renderedDepth: {
-        value: (depth || Infinity) - 1,
-        enumerable: false,
-      },
-      keyMap: {
-        value: new Map(),
-        enumerable: false,
-      },
-      depthMap: {
-        value: new Map(),
-        enumerable: false,
-      },
-      componentMap: {
-        value: new Map(),
-        enumerable: false,
-      },
-      componentNoopMap: {
-        value: new Map(),
-        enumerable: false,
-      },
-      vdomMap: {
-        value: new Map(),
-        enumerable: false,
-      },
-      fragment: {
-        value: document.createDocumentFragment(),
-        enumerable: false,
-      },
+    setHiddenProp(this, 'context', this);
+    setHiddenProp(this, 'renderedDepth', (depth || Infinity) - 1);
+    setHiddenProp(this, 'fragment', document.createDocumentFragment());
+
+    // Create our Maps
+    ['keyMap', 'depthMap', 'componentMap', 'componentNoopMap', 'vdomMap'].forEach(prop => {
+      setHiddenProp(this, prop, new Map());
     });
+
+    // Render an Empty ContextRootWrapper.  This sets up `this.contextRender`.
+    render({
+      nodeName: ContextRootWrapper,
+      attributes: {
+        vdom: null,
+        context: this,
```

Comment:
In my other method I passed in a `renderRef: render => this.contextRender = render` but it felt confusing in the construction of the wrapper this way.  Feel free to tweak or land as is and we can bikeshed later